test/test_function.rb

```diff
@@ -148,4 +148,97 @@ def call request
     startup.call "the-startup", globals: globals
     function.call "the-function", globals: globals
   end
+
+  describe "typed" do
+    # Ruby class representing a single integer value encoded as a JSON int
+    class IntValue
+      def initialize val
+        @value = val
+      end
+
+      def self.decode_json json
+        IntValue.new json.to_i
+      end
+
+      def to_json(*_args)
+        get.to_s
+      end
+
+      def get
+        @value
+      end
+    end
+
+    # class_func provides a function as a class that implements the Callable
+    # interface.
+    class_func = ::Class.new FunctionsFramework::Function::Callable do
+      define_method :call do |_request|
+        global :function_name
+      end
+    end
+
+    it "can be defined with no custom type" do
+      function = FunctionsFramework::Function.typed "int_adder" do |request|
+        request + 1
+      end
+      assert_equal "int_adder", function.name
+      assert_equal :typed, function.type
+
+      res = function.call 1, globals: {}
+
+      assert_equal 2, res
+    end
+
+    it "can be defined using a block and custom_type" do
+      function = FunctionsFramework::Function.typed "int_adder", request_class: IntValue do |request|
+        IntValue.new request.get + 1
```

Comment:
I think the precedence works here, but it's better to clarify it: `IntValue.new(request.get + 1)`

Reply:
oops, just noticed that rubocop actually goes back and removes these parentheses. The lint doesn't seem to like it though I'm inclined to agree as well that it's more clear with the parens.